src/passes/identifierManglerPass/declarationNameMangler.ts

```diff
@@ -141,5 +163,31 @@ export class DeclarationNameMangler extends ASTMapper {
     node.vFunctions.forEach((n) => this.mangleFunctionDefinition(n));
     node.vContracts.forEach((n) => this.mangleContractDefinition(n));
     this.commonVisit(node, ast);
+
+    // Checking if counter is greater than initialIdWidth digits. If so, names are
+    // modified to insert 0's to achieve id's fixed width.
+    const lastIdSize = this.lastUsedId.toString().length;
+    if (lastIdSize > this.initialIdWidth) {
+      this.nodesNameModified.forEach((node) => {
+        if (node instanceof FunctionDefinition) {
+          node.name = this.updateName(node.name, MANGLED_INTERNAL_USER_FUNCTION, lastIdSize);
+        } else if (node instanceof VariableDeclaration) {
+          node.name = this.updateName(node.name, MANGLED_LOCAL_VAR, lastIdSize);
+        } else {
+          throw new TranspileFailedError(`Not expected node to update name: ${node}`);
+        }
+      });
+    }
+    this.nodesNameModified = []; // Lose all references to avoid consuming unnecessary memory
+  }
+
+  // Set Id in a node name 'name' that fallows the pattern 'pattern' to a fixed width
```

Comment:
I am not sure I understand this commnet. fallows -> allows?

Reply:
Those node names can start with "__warp_usrfn_" or "__warp_usrid_". Let's say they are pattern1 and pattern2. If node name starts with "__warp_usrfn_" it fallows a pattern (pattern1 in this case). Then the id substring in that name is extended with 0's to get a specific length. The way to get the id substring is with regex using that pattern.